tldw_Server_API/app/core/AuthNZ/repos/identity_provider_repo.py

```
@@ -0,0 +1,486 @@
+from loguru import logger
+
+from tldw_Server_API.app.core.AuthNZ.database import DatabasePool
+
+
+def _normalize_scope_type(scope_type: str) -> str:
+    normalized = (scope_type or "").strip().lower()
+    if normalized in {"global", "system"}:
+        return "global"
+    if normalized in {"org", "organization", "organizations"}:
+        return "org"
+    raise ValueError(f"Invalid owner_scope_type: {scope_type}")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)